activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-hal-rest-api/src/test/java/org/activiti/runtime/ProcessInstanceRestTemplate.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.runtime;
+
+import java.util.Map;
+
+import org.activiti.client.model.ExtendedProcessInstance;
+import org.activiti.client.model.ProcessInstance;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+
+public class ProcessInstanceRestTemplate {
```

Comment:
Do we really need all these types? I know that the tests might look bigger, but what's the point of having all these wrappers if we can show simple tests doing pure requests to the service as any client will do. If we start adding all these types we will leak business logic to the clients and we should try to avoid doing that as much as possible. Wdyt?